internal/impl/sql/processor_sql_raw.go

```diff
@@ -213,70 +260,121 @@ func (s *sqlRawProcessor) ProcessBatch(ctx context.Context, batch service.Messag
 	s.dbMut.RLock()
 	defer s.dbMut.RUnlock()
 
-	var argsExec *service.MessageBatchBloblangExecutor
-	if s.argsMapping != nil {
-		argsExec = batch.BloblangExecutor(s.argsMapping)
+	argsExec := make([]*service.MessageBatchBloblangExecutor, len(s.queries))
+	for i, q := range s.queries {
+		if q.argsMapping != nil {
+			argsExec[i] = batch.BloblangExecutor(q.argsMapping)
+		}
+	}
+	dynQueries := make([]*service.MessageBatchInterpolationExecutor, len(s.queries))
+	for i, q := range s.queries {
+		if q.dynamic != nil {
+			dynQueries[i] = batch.InterpolationExecutor(q.dynamic)
+		}
 	}
 
 	batch = batch.Copy()
+
 	for i, msg := range batch {
```

Comment:
The logic here is a little more complex than I would like, so please do a close pass to make sure I didn't muck something up.